spec/support/locations.rb

```diff
@@ -7,30 +7,15 @@ def locations
     }
   end
 
-  # rubocop:disable Metrics/MethodLength
   def mock_location(location_name)
     location = locations[location_name.to_sym]
     page.execute_script "
-      var __bind = function(fn, me){ return function(){ return fn.apply(me, arguments); }; };
-
-      Refuge.Library.Geocoder = (function() {
-        function Geocoder() {
-          this.getAddress = __bind(this.getAddress, this);
-          this.geocodeSearchString = __bind(this.geocodeSearchString, this);
-        }
-
-        Geocoder.prototype.getCurrentLocation = function() {
-          return jQuery.when(
-            {
-              latitude: #{location[:latitude]},
-              longitude: #{location[:longitude]}
-              });
-        };
-
-        return Geocoder;
-
-      })();
+      navigator.geolocation.getCurrentPosition = function(success, failure) {
+        success({ coords: {
+          latitude: #{location[:latitude]},
+          longitude: #{location[:longitude]}
+        }, timestamp: Date.now() });
+      }
```

Comment:
Seems like this is mostly just moving to modern JS promises instead of JQuery stuff. Which is good, I suppose, perhaps not using standard promises was why this spec broke? Hmm.

Anyhow, it's working, which is the most important thing!

Reply:
I'm actually a little bit unclear why we were defining this as `Refuge.Library.Geocoder.getCurrentLocation()` before, and it's being defined as `navigator.geolocation.getCurrentPosition()` now.

But, unfortunately, I'm just not very versed in the underlying code of this web app, I've been focused on bugfixes and dependency updates. So, if this works, I'm inclined to believe it is the right thing to do.

Would like to understand better why it worked as `Refuge.Library.Geocoder.getCurrentLocation()` before, stopped working, and works as `navigator.geolocation.getCurrentPosition()` now. But I'm not sure if it's worth it to hold up the PR until this can be explained better, and my not understanding it may be down to my own unfamiliarity with the code to begin with.

I can mainly say the new code looks cleaner and more readable, that's about it. To be quite transparent.

Thank you anyway! Regardless of if you wish to try to explain this or not, the fix is appreciated either way!

Reply:
[At the time that the code was written(69d417)](https://github.com/RefugeRestrooms/refugerestrooms/commit/69d4178eacb40d7d27c0b292265d1747ddb846bd#diff-6f5ff1d87b788bedaa18b6c95d5c0f58c80e4b788afaccf35a5cdcb7ddbe560bR15), `Geocoder` was [`Refuge.Library.Geocoder`(69d417)](https://github.com/RefugeRestrooms/refugerestrooms/blob/69d4178eacb40d7d27c0b292265d1747ddb846bd/app/assets/javascripts/lib/geocoder.coffee#L3). Then It was rewritten to `Geocoder`, [applying ES6/Webpacker](https://github.com/RefugeRestrooms/refugerestrooms/commit/ee25f6af625d06269c77377a3ea594308a644de4#diff-f05e16dd1a276332c2797b2179ff36a114f229e54a470da0bae3822bd3b44665). However, this change did not cause any test failure for two reasons :

1. The poltergeist was set to ignore javascript errors with [`js_errors: false`](https://github.com/RefugeRestrooms/refugerestrooms/pull/691/files#diff-2067eb4839706ccfe3b1b74f11941c24d0c73ba9cea2f87f950cfe054e11f530L13). So you could replace the code with anything like `asdf.qwer.blah()`.
2. The tests using `mock_location` in `restroom_spec.rb` never seems to have been written completely since https://github.com/RefugeRestrooms/refugerestrooms/commit/cd52015fcb7addc62ecd4440956a97f70663dc40 except for one test that [didn't require an exact location](https://github.com/RefugeRestrooms/refugerestrooms/commit/cd52015fcb7addc62ecd4440956a97f70663dc40#diff-24052ed581e8aa075482e4a55dab66f682a33b33bb8a43d94856dde11151e5aeR49). Some of them just wrapped with [`#rubocop:disable RSpec/NoExpectationExample`](https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecnoexpectationexample). So I think return values of `getCurrentLocation()` would never be a problem. I'll try to complete them later.

Thanks for review this PR.